src/openrct2/core/String.cpp

```diff
@@ -338,7 +338,15 @@ namespace OpenRCT2::String
         // null terminator.
         va_list copy;
         va_copy(copy, args);
+#pragma GCC diagnostic push
+// clang does not recognize -Wformat-truncation and errors on -Wunknown-warning-option.
+// GCC does not recognize disabling -Wunknown-warning-option via pragma
+// Only disable this warning for GCC, as Clang does not support it.
+#ifndef __clang__
+    #pragma GCC diagnostic ignored "-Wformat-truncation"
+#endif
         auto len = vsnprintf(nullptr, 0, format, copy);
```

Comment:
Wouldn't it be better to actually supply a buffer of lets say `char[128]` and see if it worked, if the length returned is more or equal the size of the buffer only then we would have to branch out to create a bigger buffer, this would be also a minor performance optimization given it doesn't have to first count.